pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1148,56 +1143,76 @@ func taintX509CA(tx *gorm.DB, trustDomainID string, taintedPublicKey crypto.Publ
 	return nil
 }
 
-func revokeX509CA(tx *gorm.DB, trustDomainID string, publicKey crypto.PublicKey) error {
+func revokeX509CA(tx *gorm.DB, trustDomainID string, publicKeyToRevoke crypto.PublicKey) error {
 	bundle, err := getBundle(tx, trustDomainID)
 	if err != nil {
 		return err
 	}
 
-	found := false
+	var taintedKeyFound bool
+	var taintedKeys []*common.X509TaintedKey
+
+	for _, eachTaintedKey := range bundle.X509TaintedKeys {
+		taintedKey, err := x509.ParsePKIXPublicKey(eachTaintedKey.PublicKey)
+		if err != nil {
+			return status.Errorf(codes.Internal, "failed to parse tainted Key: %v", err)
+		}
+
+		ok, err := cryptoutil.PublicKeyEqual(taintedKey, publicKeyToRevoke)
+		if err != nil {
+			return status.Errorf(codes.Internal, "failed to compare public key: %v", err)
+		}
+		if ok {
+			taintedKeyFound = true
+			continue
+		}
+
+		taintedKeys = append(taintedKeys, eachTaintedKey)
+	}
+
+	if !taintedKeyFound {
+		return status.Error(codes.InvalidArgument, "it is not possible to revoke an untainted root CA")
```

Comment:
Good check! I think it may be better to handle in local authority api though?

Reply:
I was not sure about that, I'm ok in moving it to service layer, but revoking on datastore without this validation worry me

Reply:
Ok .. it felt weird to me for datastore to be knowing about this kind of thing, but I think it's fine to leave as a defensive measure 👍